web/src/App.js

```diff
@@ -99,9 +107,12 @@ function App() {
               />
             </Category>
 
-            { isInstalling &&
+            { isInstalling && progress &&
               <Category title="Progress" icon={Clock} >
-                <Progress hasStripe value={Math.round(parseInt(installation.status) / STEPS * 100)} />
+                <Text>Installing</Text>
+                <Progress hasStripe value={Math.round(progress.step / progress.steps * 100)} />
+                <Text>{ progress.title }</Text>
+                <Progress hasStripe value={Math.round(progress.substep / progress.substeps * 100)} />
```

Comment:
this can be conditional if substeps > 0 . In other cases it is not needed. Not sure what will look better.

Reply:
Yes, it can be. Most probably, we might need to separate this piece of code into a separate component.